src/Components/Search/NewSearch/NewSearchBarInput.tsx

```diff
@@ -10,92 +9,213 @@ import { NewSearchBarInput_viewer$data } from "__generated__/NewSearchBarInput_v
+
+        setFetchCounter(prevCounter => prevCounter + 1)
       }
     )
   }
 
+  const debouncedSearchRequest = useMemo(
+    () => debounce(refetch, SEARCH_DEBOUNCE_DELAY),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    []
+  )
+
+  const clearSearchInput = () => {
+    setValue("")
+  }
+
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setValue(event.target.value)
+
+    if (event.target.value.length > 1)
+      debouncedSearchRequest(event.target.value)
+  }
+
+  const handlePillClick = (pill: PillType) => {
+    setSelectedPill(pill)
+    refetch(value, pill.searchEntityName)
+  }
+
+  const handleRedirect = () => {
+    clearSearchInput()
+  }
+
+  const handleSubmit = () => {
+    if (value) {
+      router.push(`/search?term=${encodeURIComponent(value)}`)
+    }
+  }
+
+  const handleSelect = (option: SuggionItemOptionProps) => {
+    tracking.trackEvent({
+      action_type: DeprecatedSchema.ActionType.SelectedItemFromSearch,
+      destination_path:
+        option.typename === "Artist"
+          ? `${option.href}/works-for-sale`
+          : option.href,
+      item_number: option.item_number,
+      item_type: option.item_type,
+      query: value,
+    })
+
+    clearSearchInput()
+    router.push(option.href)
+  }
+
+  const handleFocus = () => {
+    tracking.trackEvent({
+      action_type: DeprecatedSchema.ActionType.FocusedOnAutosuggestInput,
+    })
+  }
+
   return (
     <AutocompleteInput
       placeholder={isXs ? t`navbar.searchArtsy` : t`navbar.searchBy`}
       spellCheck={false}
       options={value.length < 2 ? [] : formattedOptions}
       value={value}
       onChange={handleChange}
-      onClear={() => setValue("")}
-      header={<NewSearchInputPillsFragmentContainer viewer={viewer} />}
+      onClear={clearSearchInput}
+      onSubmit={handleSubmit}
+      onFocus={handleFocus}
+      onSelect={handleSelect}
+      header={
+        <NewSearchInputPillsFragmentContainer
+          viewer={viewer}
+          selectedPill={selectedPill}
+          onPillClick={handlePillClick}
+        />
+      }
       renderOption={option => (
-        <SuggestionItem
-          display={option.text}
-          href={option.href}
-          isHighlighted={false} // TODO: change
-          label={option.subtitle ?? ""}
-          imageUrl={option.imageUrl}
+        <NewSuggestionItem
           query={value}
-          showArtworksButton={option.showArtworksButton}
-          showAuctionResultsButton={option.showAuctionResultsButton}
+          option={option}
+          onRedirect={handleRedirect}
         />
       )}
       footer={
         <NewSearchBarFooter
-          display={value}
-          href={`/search?term=${encodeURIComponent(value)}`}
-          isHighlighted={false}
-          label={value}
           query={value}
+          href={`/search?term=${encodeURIComponent(value)}`}
+          onRedirect={handleRedirect}
+          currentIndex={options.length}
```

Comment:
**suggestion**: use `index` instead of `currentIndex` (feel free to ignore if you disagree)